packages-backend/express/src/auth.ts

```diff
@@ -36,47 +78,59 @@ const throwIfIssuerIsNotAValidUrl = (issuer: string) => {
     );
   }
 };
-
-type TDecodedJWT = {
-  sub: string;
-  iss: string;
-  [property: string]: string;
+// Validates required option values.
+const validateRequiredValues = (options: TSessionMiddlewareOptions) => {
+  if (!options.audience) {
+    throw new Error(`Missing required option "audience"`);
+  }
+  if (!options.issuer) {
+    throw new Error(`Missing required option "issuer"`);
+  }
 };
-
-const writeSessionContext = <Request extends IncomingMessage>(
-  request: Request & { decoded_token: TDecodedJWT; session?: TSession }
-) => {
-  const decodedToken = request[decodedTokenKey];
-  const publicClaimForProjectKey = `${decodedToken.iss}/claims/project_key`;
-
-  request.session = {
-    userId: decodedToken.sub,
-    projectKey: decodedToken[publicClaimForProjectKey],
-  };
-
-  delete request.decoded_token;
+// Attempt to parse the given issuer. If the value is a cloud identifier, it will
+// be mapped to one of the supported values. If not, we assume the value is a valid URL.
+const parseDefaultIssuer = (options: TSessionMiddlewareOptions) => {
```

Comment:
Is this parsing or just mapping. Just trying to follow along with the code. `inferIssuerDefaultOrThrow`? It becomes odd :)

Reply:
This function should return a value, either from the cloud identifier or the manually passed url.
I couldn’t come up with a better name